src/async_monitored_item.rs

```diff
@@ -222,21 +217,50 @@ impl MonitoredItemBuilder {
     }
 }
 
+/// Value emitted from monitored item notification.
+///
+/// The variant depends on the attribute ID passed to [`MonitoredItemBuilder::attribute_id()`].
+#[derive(Debug)]
+pub enum MonitoredItemValue {
+    /// Data change payload.
+    ///
+    /// This is emitted for attribute IDs other than [`ua::AttributeId::EVENTNOTIFIER`].
+    DataChange(ua::DataValue),
```

Comment:
Should we use explicit variant structs instead of an anonymous tuple structs here?

```
DataChange { value: ua::DataValue }
Event { fields: ua::Array<ua::Variant> }
```

Easier to understand for me who was not familiar with both concepts.